xds/src/main/java/io/grpc/xds/Bootstrapper.java

```diff
@@ -35,58 +35,35 @@
 import java.util.Map;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
-import javax.annotation.concurrent.ThreadSafe;
 
 /**
  * Loads configuration information to bootstrap gRPC's integration of xDS protocol.
  */
-@ThreadSafe
 abstract class Bootstrapper {
 
   private static final String BOOTSTRAP_PATH_SYS_ENV_VAR = "GRPC_XDS_BOOTSTRAP";
 
-  static Bootstrapper newInsatnce() {
-    return new FileBasedBootstrapper();
+  private static final Bootstrapper DEFAULT_INSTANCE = new Bootstrapper() {
+    @Override
+    BootstrapInfo readBootstrap() throws IOException {
+      String filePath = System.getenv(BOOTSTRAP_PATH_SYS_ENV_VAR);
+      if (filePath == null) {
+        throw new IOException("Environment variable " + BOOTSTRAP_PATH_SYS_ENV_VAR + " not found.");
+      }
+      return parseConfig(
+          new String(Files.readAllBytes(Paths.get(filePath)), StandardCharsets.UTF_8));
+    }
+  };
+
+  static Bootstrapper newInstance() {
```

Comment:
Misleading name: this one returns a single instance that was statically created. You can call it `getInstance` to avoid the wrong impression.

Reply:
Changed.